File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/nightstudy/exception/NightStudyExceptionCode.java
package b1nd.dodam.domain.rds.nightstudy.exception;

import b1nd.dodam.core.exception.ExceptionCode;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public enum NightStudyExceptionCode implements ExceptionCode {

    INVALID_STUDY_DURATION(400, "잘못된 심야자습 기간"),
    REASON_FOR_PHONE_MISSING(400, "핸드폰 사용이유 누락"),
    APPLICATION_DURATION_PASSED(403, "심야자습 신청 기간이 아님"),
    NOT_APPLICANT(403, "신청자가 아님"),
    NOT_FOUND(404, "없는 심야자습"),
    STUDY_DUPLICATE(409, "해당 날짜에 심야자습 중복");

    private final int status;
    private final String message;

    @Override
    public int getStatus() {
        return this.status;
    }

    @Override
    public String getExceptionName() {
        return this.name();
    }

    @Override
    public String getMessage() {
        return this.message;
    }

}

File: services/service-night-study/src/main/kotlin/com/b1nd/dodamdodam/nightstudy/domain/nightstudy/exception/NightStudyExceptionCode.kt
package com.b1nd.dodamdodam.nightstudy.domain.nightstudy.exception

import com.b1nd.dodamdodam.core.common.exception.ExceptionCode
import org.springframework.http.HttpStatus

enum class NightStudyExceptionCode(
    override val status: HttpStatus,
    override val message: String
) : ExceptionCode {
    NIGHT_STUDY_NOT_FOUND(HttpStatus.NOT_FOUND, "심야자습을 찾을 수 없어요."),
    NIGHT_STUDY_NOT_OWNER(HttpStatus.FORBIDDEN, "본인의 심야자습만 삭제할 수 있어요."),
    NIGHT_STUDY_BANNED(HttpStatus.FORBIDDEN, "심야자습이 정지된 상태에요."),
    ;
}
